api/src/factorydb/Questionsdb.js

```diff
@@ -0,0 +1,30 @@
+import sqlite3 from "sqlite3";
+import { ErrorCheckDB } from "../../components/ErrorHandling/ErrorCheckDB.js";
+
+export let question_db = new sqlite3.Database("./database/Questions.db", (err) => {
+  ErrorCheckDB(err, "Questions", "Questions Database Connection error");
+  console.log("Connected To the Questions SQLite db");
+  question_db.run(
+    `CREATE TABLE IF NOT EXISTS Questions (
+      id INTEGER PRIMARY KEY AUTOINCREMENT,
+      userId INTEGER NOT NULL,
+      question TEXT NOT NULL,
+      FOREIGN KEY (userId) REFERENCES Userdb(id)
+    )`,
+    (err) => {
+      ErrorCheckDB(err, "Questions", "Error creating table");
+      question_db.get("SELECT COUNT(*) AS count FROM Questions", (err, row) => {
+        ErrorCheckDB(err, "Questions", "Error checking table count");
+        if (row.count === 0) {
+          console.log("Questions Table is ready. Inserting initial data.");
+          let insert = "INSERT INTO Questions (userId, question) VALUES (?, ?)";
```

Comment:
This is good. Include their username into the sql here, also include the username for both of the tables as well

Reply:
but i think we can still do that with foreign keys i think